cloudferrylib/os/image/glance_image.py

```diff
@@ -196,14 +196,23 @@ def get_image_list(self):
             for f in self.get_image_filter().get_filters():
                 images = ifilter(f, images)
             images = [i for i in images]
-
-            LOG.info("Filtered images: %s",
-                     ", ".join((str(i.name) for i in images)))
+        else:
+            images = [img for img in images if not img.deleted]
+        LOG.info("Filtered images: %s",
+                 ", ".join(('%s (%s)' % (i.name, i.id) for i in images)))
 
         return images
 
     def create_image(self, **kwargs):
-        return self.glance_client.images.create(**kwargs)
+        try:
+            return self.glance_client.images.create(**kwargs)
+        except glance_exceptions.HTTPConflict:
+            image_id = kwargs.pop('id', None)
+            if image_id is not None:
+                LOG.debug("Retry to create an image '%s' (%s) without old id",
```

Comment:
Let's have this message as warning, and change message to something like: "Image ID will not be kept for source image %(id)s, image ID is already present in destination (perhaps was deleted previously)."